File: src/main/java/com/example/accounting/model/Transaction.java
package com.example.accounting.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    private Account account;
    private TransactionDirection direction;
    private BigDecimal amount;
    private LocalTime time;
}

File: backend/src/main/kotlin/com/ptah/entity/accounting/Transaction.kt
package com.ptah.entity.accounting

import com.ptah.common.BaseEntity
import jakarta.persistence.*
import java.math.BigDecimal


@Entity
class Transaction(account: Account, debit: TransactionDirection, amount: BigDecimal) : BaseEntity() {
    @ManyToOne
    val account: Account? = null

    @Enumerated(EnumType.STRING)
    val direction: TransactionDirection? = null
    val amount: BigDecimal? = null
}
